src/preferences/configobject.h

```diff
@@ -171,6 +178,13 @@ template <class ValueType> class ConfigObject {
     template <class ResultType>
     ResultType getValue(const ConfigKey& key, const ResultType& default_value) const;
     QString getValue(const ConfigKey& key, const char* default_value) const;
+    template<typename EnumType>
+    requires std::is_enum_v<EnumType>
```

Comment:
```suggestion
    template <class ResultType, std::enable_if_t<!std::is_enum_v<ResultType>, bool> = true>
    ResultType getValue(const ConfigKey& key, const ResultType& default_value) const;
    QString getValue(const ConfigKey& key, const char* default_value) const;
    template<typename EnumType, std::enable_if_t<std::is_enum_v<EnumType>, bool> = true>
```

Reply:
explicitly opting out of enums for the default template is unnecessary when using `requires` as the template with `requires` has a higher specificity and is thus chosen over the unconstrained template. I'd prefer to not have the negative requirement, as the maintenance burden will explode in case we add even more templates.